docs/basics/networkbehaviour.md

```diff
@@ -108,3 +108,130 @@ If you override the virtual 'OnDestroy' method it is important to alway invoke t
+- All `NetworkObject` oriented synchronization information is deserialized
+  - Dynamically spawned `NetworkObject`s are instantiated and state is synchronized
+  - For each `NetworkObject` instance:
+    - `NetworkVariable` state is deserialized and applied.
+    - `NetworkBehaviour.OnSynchronize` is invoked.
+      - If this method is not overridden then nothing is read from the serialization buffer.
+    - The `NetworkObject` is spawned
+      - For each associated `NetworkBehaviour` component, `NetworkBehaviour.OnNetworkSpawn` is invoked.
+
+### OnSynchronize Example 
+Now that you understand the general concept behind `NetworkBehaviour.OnSynchronize`, the question you might have is "when would you use such a thing"? `NetworkVariable`s can be useful to synchronize state, but they also are only updated every network tick and you might have some form of state that needs to be updated when it happens and not several frames later so you decide to use RPCs.  However, this becomes an issue when you want to synchronize late joining clients as there is no way to really synchronize late joining clients based on RPC activity over the duration of a network session.  This is one of many possible reasons one might want to use `NetworkBehaviour.OnSynchronize`.
+In the example below, it provides one simple use-case scenario where you can use `NetworkBehaviour.OnSynchronize` to synchronize late-joining clients with state set by `ClientRpc` events:
+```csharp
+using UnityEngine;
+using Unity.Netcode;
+
+/// <summary>
+/// Simple RPC driven state that demonstrates one
+/// form of NetworkBehaviour.OnSynchronize usage
+/// </summary>
+public class SimpleRpcState : NetworkBehaviour
+{
+    private bool m_ToggleState;
+
+    /// <summary>
+    /// Late joining clients will be synchronized 
+    /// to the most current m_ToggleState
+    /// </summary>
+    protected override void OnSynchronize<T>(ref BufferSerializer<T> serializer)
+    {
+        serializer.SerializeValue(ref m_ToggleState);
+        base.OnSynchronize(ref serializer);
+    }
+
+    public void ToggleState(bool stateIsSet)
+    {
+        m_ToggleState = stateIsSet;
+    }
+
+    /// <summary>
+    /// Synchronizes connected clients with the
+    /// server-side m_ToggleState
+    /// </summary>
+    /// <param name="stateIsSet"></param>
+    [ClientRpc]
+    private void ToggleStateClientRpc(bool stateIsSet)
+    {
+        m_ToggleState = stateIsSet;
+    }
+}
+```
+:::caution
+Since `NetworkBehaviour.OnSynchronize` is primarily used for server to client synchronization, RPC state synchronization only works when using ClientRpcs since `NetworkBehaviour.OnSynchronize` is only invoked on the server side during the write portion of serialization and only invoked on the client side during the read portion of serialization. When running a host `NetworkBehaviour.OnSynchronize` is still only invoked once (server-side) during the write portion of serialization.
+:::
+
```

Comment:
Now that you understand the general concept behind `NetworkBehaviour.OnSynchronize`, you might wonder when you would use such a thing. 

You can use `NetworkVariable`s to synchronize state, but they only update every network tick. If you have some form of state that you need to update when it happens (and not several frames later), you might consider using RPCs instead of `NetworkVariable's. 

However, RPCs become an issue when you want to synchronize late-joining clients because there's no way to synchronize late-joining clients based on RPC activity throughout a network session. In such a scenario, you can use `NetworkBehaviour.OnSynchronize`.

The example below provides one simple use-case scenario where you can use `NetworkBehaviour.OnSynchronize` to synchronize late-joining clients with state set by `ClientRpc` events:

// CODE

:::caution
Since `NetworkBehaviour.OnSynchronize` is primarily used for server-to-client synchronization, RPC state synchronization only works when using `ClientRpcs` because `NetworkBehaviour.OnSynchronize` is only invoked on the server-side during the write portion of serialization and only invoked on the client-side during the read portion of serialization. When running a host, `NetworkBehaviour.OnSynchronize` only invokes once (server-side) during the write portion of serialization.
:::